File: Bai_49/main.py
class Rectangle:
    def __init__(self, length, width):
        self.length = length
        self.width = width

    def calculate_area(self):
        return self.length * self.width


class Trapezoid:
    def __init__(self, base1, base2, height):
        self.base1 = base1
        self.base2 = base2
        self.height = height

    def calculate_area(self):
        return (self.base1 + self.base2) * self.height / 2


rectangle = Rectangle(5, 8)
trapezoid = Trapezoid(3, 7, 4)

if rectangle.calculate_area() > trapezoid.calculate_area():
    print("Rectangle has a larger area.")
elif rectangle.calculate_area() < trapezoid.calculate_area():
    print("Trapezoid has a larger area.")
else:
    print("Both shapes have equal areas.")

File: Bai_49/main.cpp
#include <iostream>
using namespace std;

class Rectangle {
public:
    Rectangle(double length, double width) : length(length), width(width) {}

    double calculateArea() {
        return length * width;
    }

private:
    double length;
    double width;
};

class Trapezoid {
public:
    Trapezoid(double base1, double base2, double height) : base1(base1), base2(base2), height(height) {}

    double calculateArea() {
        return (base1 + base2) * height / 2;
    }

private:
    double base1;
    double base2;
    double height;
};

int main() {
    Rectangle rectangle(5, 8);
    Trapezoid trapezoid(3, 7, 4);

    if (rectangle.calculateArea() > trapezoid.calculateArea()) 
        cout << "Rectangle has a larger area." << endl;
    else if (rectangle.calculateArea() < trapezoid.calculateArea()) 
        cout << "Trapezoid has a larger area." << endl;
    else 
        cout << "Both shapes have equal areas." << endl;

    return 0;
}
